pygti/schemas.py

```diff
@@ -84,7 +95,74 @@
         "filter": [FilterEntry],
         "serviceTypes": [FilterServiceType],
         "useRealtime": bool,
+    },
+)
+
+SIRequest = Schema.extend(BaseRequestType, {"station": SDName})
+
+ContSearchByServiceId = Schema(
+    {
+        "serviceId": int,
+        "lineKey": str,
+        "plannedDepArrTime": GTITime,
+        "additionalOffset": int,
     }
 )
 
-SIRequest = Schema({"station": SDName})
+# kinds: GTI Docs say int but traffic on the website uses strings
+# Same thing with the value of the penalty. All are strings
+TariffInfoSelector = Schema({"tariff": str, "tariffRegions": bool, "kinds": [str],})
+PenaltyName = In(
+    [
+        "ChangeEvent",
+        "ExtraFare",
+        "Walker",
+        "AnyHandicap",
+        "ToStartStationBy",
+        "TimeRange",
+        "ForVisitors",
+        "desiredType",
```

Comment:
DesiredType

Reply:
sadly not. Should be DesiredType if the API would be consistent, but it's not...
DesiredType will give you an error